src/custom/utils/metadata.ts

```diff
@@ -0,0 +1,51 @@
+export enum MetadataKind {
+  REFERRAL = 'referrer',
+}
+
+interface Metadata {
+  kind: MetadataKind
+  version: string
+}
+
+export interface ReferralMetadata extends Metadata {
+  referrer: string
+}
+
+export type MetadataDoc = {
+  [MetadataKind.REFERRAL]?: ReferralMetadata
+}
+
+export type AppDataDoc = {
+  version: string
+  appCode?: string
+  metadata: MetadataDoc
+}
+
+export const DEFAULT_APP_CODE = 'CowSwap'
+
+export function generateReferralMetadataDoc(
```

Comment:
An alternative to this method would be, to expose `generateReferralMetadata` instead of `generateReferralMetadataDoc`

Then the one logic creating the document would just do something like:

```ts
const referralMetadata = generateReferralMetadata(referralAddress)

const appDataDoc = generateAppDataDoc({
 appCode: DEFAULT_APP_CODE,
 metadata: [ referralMetadata ] 
 })
```